From 182c58160d32c27e56b43b66efb686589ad852c8 Mon Sep 17 00:00:00 2001
From: Philip Mueller <philm@manjaro.org>
Date: Fri, 31 Jul 2026 21:53:22 +0200
Subject: [PATCH 042/176] [FROM-ML] iio: imu: bmi270: Match PNP ID found on
 OrangePi Neo 2026

OrangePi Neo devices use BMI260 sensors in their hardware.
In older firmware they used "BMI0260" as their ID. In newer firmware
however they switched to "BOSC0260" as their ID for the sensors.
This change is also seen DSDT tables within newer device firmware.

Adding the PNP ID to bmi270 driver will load the driver properly on the device.

Cc: stable@vger.kernel.org # 6.18+
Signed-off-by: Philip Mueller <philm@manjaro.org>
---
 drivers/iio/imu/bmi270/bmi270_i2c.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/iio/imu/bmi270/bmi270_i2c.c b/drivers/iio/imu/bmi270/bmi270_i2c.c
index 7c035e82e6e9..036c6f0f7cd4 100644
--- a/drivers/iio/imu/bmi270/bmi270_i2c.c
+++ b/drivers/iio/imu/bmi270/bmi270_i2c.c
@@ -43,6 +43,8 @@ static const struct acpi_device_id bmi270_acpi_match[] = {
 	{ "BMI0160",  (kernel_ulong_t)&bmi260_chip_info },
 	/* GPD Win Max 2 2023(sincice BIOS v0.40), etc. */
 	{ "BMI0260",  (kernel_ulong_t)&bmi260_chip_info },
+	/* OrangePi Neo 2026 (since BIOS BCPH01_V1.20), etc. */
+	{ "BOSC0260", (kernel_ulong_t)&bmi260_chip_info },
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, bmi270_acpi_match);
-- 
2.55.0

